src/validate.ts

```
@@ -1,113 +1,33 @@
-}
+const ajv = new Ajv({
+  allErrors: true,
+});
 
 /**
- * Generates EXECUTE request 
- * @param devices Array of type Device containing device IDs and respective custom data 
- * @param execution Array of type Command specifying the commands of the devices and their respective parameters 
- * @returns Specified format for EXECUTE intent request. 
+ * Helper function that uses AJV library to validate the response against the schema 
+ * @param apiResponse User defined api response
+ * @return Returns undefined if valid is true, returns an array of error(s) if valid is false. 
  */
-export function generateExecuteRequest(devices: Device[], execution: Command[]): ExecuteRequest{
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.EXECUTE",
-          payload: {
-            devices,
-            execution
-          }
-        }]
-    }
+function validateSyncResponse(apiResponse: object) {
+  const syncResponseSchema = require('../intents/sync.response.schema.json');
+  const isValid = ajv.validate(syncResponseSchema, apiResponse);
+  if (isValid) {
+    return undefined;
+  }
+  return ajv.errors;
 }
 
 /**
- * Generates DISCONNECT request 
- * @returns Specified format for DISCONNECT intent request. 
+ * Identifies the response type and validates the function based on the schemas.
+ * @param apiResponse User defined api response 
+ * @param responseType User defined intent response 
+ * @return Errors from AJV validation, if any. Undefined otherwise. 
  */
-export function generateDisconnectRequest(){
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.DISCONNECT"
-        }]
-    }
+export function validate(apiResponse: object, responseType: 'sync' | 'query'|'execute'| 'disconnect') {
+  if (responseType == 'sync') {
+    return validateSyncResponse(apiResponse);
+  }
+  return undefined;
```

Would there be an error here rather than `undefined`, as an invalid `responseType` was selected.

---

Yes that's true. Since the types of responses are specified in the function, it'll throw an error automatically without specifying. But, maybe instead of returning undefined a better way would be to throw new Error(); instead? Lmk your thoughts